binary/memory.go

```diff
@@ -23,7 +23,7 @@ func decodeMemory(r *bytes.Reader) (*wasm.Memory, error) {
 		mem.IsMaxEncoded = true
 	}
 
-	if min > mem.Max {
+	if mem.Max != 0 && min > mem.Max {
```

Comment:
oh I see the bug. can you instead move the min > mem.Max into the above block instead? this should only be compared when max is encoded.

Reply:
Yep. Make sense.